File: backend/app/schemas/persona.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class PersonaCreate(BaseModel):
    name: str
    language_id: UUID
    gender: str
    type: str
    teaching_style: str | None = None


class PersonaUpdate(BaseModel):
    name: str | None = None
    gender: str | None = None
    type: str | None = None
    teaching_style: str | None = None
    image_url: str | None = None
    is_active: bool | None = None


class PersonaResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    name: str
    language_id: UUID
    image_url: str | None
    idle_video_url: str | None = None
    gender: str
    type: str
    teaching_style: str | None
    is_active: bool
    created_at: datetime


class PersonaWithStatusResponse(PersonaResponse):
    idle_loop_status: str = "none"

    @classmethod
    def from_persona(cls, persona: "PersonaResponse") -> "PersonaWithStatusResponse":
        data = persona.model_dump()
        if data.get("idle_video_url"):
            data["idle_loop_status"] = "ready"
        elif data.get("image_url"):
            data["idle_loop_status"] = "pending"
        else:
            data["idle_loop_status"] = "none"
        return cls(**data)

File: mobile/lib/models/persona.dart
class Persona {
  final String id;
  final String name;
  final String languageId;
  final String? imageUrl;
  final String gender;
  final String type;
  final String? teachingStyle;
  final bool isActive;

  const Persona({
    required this.id,
    required this.name,
    required this.languageId,
    this.imageUrl,
    required this.gender,
    required this.type,
    this.teachingStyle,
    this.isActive = true,
  });

  factory Persona.fromJson(Map<String, dynamic> json) {
    return Persona(
      id: json['id'] as String,
      name: json['name'] as String,
      languageId: json['language_id'] as String,
      imageUrl: json['image_url'] as String?,
      gender: json['gender'] as String,
      type: json['type'] as String,
      teachingStyle: json['teaching_style'] as String?,
      isActive: json['is_active'] as bool? ?? true,
    );
  }
}
